lib/perl/Genome/Model/Tools/Bsmap/MethylationRatio.pm

```diff
@@ -0,0 +1,99 @@
+package Genome::Model::Tools::Bsmap::MethylationRatio;
+
+use strict;
+use warnings;
+use Cwd;
+use Genome;
+
+use Genome::Utility::Text qw(sanitize_string_for_filesystem);
+use File::Spec;
+use File::Basename qw();
+
+my ( undef, $dir ) = File::Basename::fileparse(__FILE__);
+my $METHRATIO_COMMAND = File::Spec->join( $dir, 'methylation_ratio.py' );
+
+class Genome::Model::Tools::Bsmap::MethylationRatio {
+	is  => 'Command',
+	has => [
+		bam_file => {
+			is       => 'Text',
```

Comment:
Instead of `Text` you can use `FilePath` just as a form of metadata that these are intended to be actual paths.